components/post_view/post_pre_header/post_pre_header.tsx

```diff
@@ -0,0 +1,136 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {FormattedMessage, injectIntl, IntlShape} from 'react-intl';
+
+import {t} from 'utils/i18n';
+
+import FlagIconFilled from 'components/widgets/icons/flag_icon_filled';
+
+export type Props = {
+    intl: IntlShape;
+    isFlagged: boolean;
+    isPinned: boolean;
+    skipPinned?: boolean;
+    skipFlagged?: boolean;
+    channelId: string;
+    actions: {
+        showFlaggedPosts: () => void;
+        showPinnedPosts: (channelId: string) => void;
+    };
+}
+
+export enum PostPinnedOrFlagged {
+    Flagged,
+    Pinned,
+    PinnedAndFlagged,
+    Neither,
+}
+
+enum MessageInfoKey {
+    Flagged = 'flagged',
+    Pinned = 'pinned',
+    PinnedAndFlagged = 'pinnedAndFlagged',
+}
+
+class PostPreHeader extends React.PureComponent<Props> {
+    messageInfos = {
+        flagged: {id: t('post_pre_header.flagged'), defaultMessage: 'Saved'},
+        pinned: {id: t('post_pre_header.pinned'), defaultMessage: 'Pinned'},
+    };
+
+    getPostStatus(isFlagged: boolean, isPinned: boolean): PostPinnedOrFlagged {
+        if (isFlagged && isPinned) {
+            return PostPinnedOrFlagged.PinnedAndFlagged;
+        } else if (isFlagged) {
+            return PostPinnedOrFlagged.Flagged;
+        } else if (isPinned) {
+            return PostPinnedOrFlagged.Pinned;
+        }
+
+        return PostPinnedOrFlagged.Neither;
+    }
+
+    getMessageInfo(postStatus: PostPinnedOrFlagged, skipFlagged?: boolean, skipPinned?: boolean): MessageInfoKey | false {
+        if (skipFlagged && skipPinned) {
+            return false;
+        }
+
+        if (postStatus === PostPinnedOrFlagged.PinnedAndFlagged) {
+            if (!skipPinned && !skipFlagged) {
```

Comment:
Same here (no need for else's)

Reply:
Same as above, done but one-line if statements weren't possible